hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataLogRecordReader.java

```diff
@@ -106,10 +107,10 @@ public Map<String, HoodieRecord<HoodieMetadataPayload>> getRecordsByKeys(List<St
     //       materialized state, to make sure there's no concurrent access
     synchronized (this) {
       logRecordScanner.scanByFullKeys(keys);
-      return logRecordScanner.getRecords().entrySet()
-          .stream()
+      Map<String, HoodieRecord> allRecords = logRecordScanner.getRecords();
```

Comment:
What is the fix here? Previously we are iterating on entrySet any reason why we changed the iteration to keySet?

Reply:
We were not filtering on `keys` which contain valid partitions. Due to that result from this method also had data for `__all_partitions__` key which is not needed.